docs/blueprints/forgotten-deletes.md

```diff
@@ -0,0 +1,118 @@
+# Forgotten Deletes
+
+## Abstract
+
+In the event of a node being audited for a piece which the satellite previously asked it to delete via delete pieces request, the node will present a signed message proving the legitimacy of the deletion. The node is not penalized and the object is deleted. We eliminate the potential of forgotten garabage collection deletes entirely.
+
+## Background
+
+In the event of a DB restoration from backup, it may be possible for the satellite to restore pointers for objects which were deleted, or revert pointers to a previous state containing nodes which no longer have those pieces. In this situation, the satellite would audit nodes for pieces which they were told to delete which may result in unfair disqualifications.
+
+## Design
+
+The solution to this problem is twofold, as there are two ways in which pieces are deleted: delete pieces requests and garbage collection.
+
+### Delete pieces requests
+
+Delete pieces requests are issued when a user deletes an object. The metainfo service deletes the object's pointers and then sends delete pieces requests to each of the reliable nodes which held pieces. The rest of the nodes' pieces will be eventually deleted by garbage collection. 
+
+To prove to the satellite that a piece was deleted legitimately, we will have it sign the delete pieces requests. The node will present these to the satellite in the event of an audit for a deleted piece.
+
+```
+message DeletePiecesRequest {
+    repeated bytes piece_ids = 1 [(gogoproto.customtype) = "PieceID", (gogoproto.nullable) = false];
+    bytes satellite_signature = 2;
+}
+```
```

Comment:
i'm worried about performance of this table. we just barely finished ripping out the order table and replacing it with append-only files for performance. i think we should try and avoid making the same problems we had with the order table. can we benchmark a few different storage backends for put storing delete piece requests and finding them later? storing them will be frequent, but finding them should be *very* infrequent, so maybe we can make puts fast at the expense of slow gets?